### ISSUE TITLE
no-unused-vars false positive in Typescript functions overloading

### ISSUE THREAD

**Issue by steelbrain** [oxc-project/oxc#5563]
```
  × eslint(no-unused-vars): Variable 'args' is declared but never used.
    ╭─[projects/.../auth/hasUserAuthScope.ts:14:27]
 13 │ function hasUserAuthScope(...args: ArgsOne): boolean
 14 │ function hasUserAuthScope(...args: ArgsTwo): boolean
    ·                           ─────────────────┬─────────────────
    ·                                            ╰── 'args' is declared here
 15 │ function hasUserAuthScope(...args: ArgsOne | ArgsTwo): boolean {
    ╰────
  help: Consider removing this declaration.
```

Running oxlint version `0.9.2`

The relevant config is as follows

```
    "eslint/no-unused-vars": [
      "error",
      {
        "argsIgnorePattern": "^_",
        "varsIgnorePattern": "^_",
        "caughtErrors": "none"
      }
    ],
```

**Comment by camc314** [comment]
duplicate of https://github.com/oxc-project/oxc/issues/5406 ?

**Comment by steelbrain** [comment]
You're right. I didn't find a relevant *open* issue so decided to open a new one. Thanks!

Waiting on a release to include the bugfix so we can upgrade internally

**Comment by camc314** [comment]
all good. please report any other issue you find!
